From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Wed, 8 Jun 2022 14:23:32 +0800
Subject: [PATCH] fs/ntfs3: Fix the build issue after sync 5.19 patches
Patch-mainline: Never, ntfs3 is poorly supported
Modified-by-SEL: N/A
References: git-fix

Lots of errors happen after porting latest mainline patches, let's fix
them by refer openEuler patch.

patches.openEuler/14032-fs-ntfs3-Fix-the-issue-from-backport-5.15-to-5.10.patch

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/ntfs3/xattr.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/fs/ntfs3/xattr.c b/fs/ntfs3/xattr.c
index 0d77a9da54a3..4eb42b3fed75 100644
--- a/fs/ntfs3/xattr.c
+++ b/fs/ntfs3/xattr.c
@@ -629,7 +629,7 @@ static int ntfs_xattr_get_acl(struct user_namespace *mnt_userns,
 		return -EOPNOTSUPP;
 	}
 
-	acl = ntfs_get_acl(inode, type, false);
+	acl = ntfs_get_acl(inode, type);
 	if (IS_ERR(acl))
 		return PTR_ERR(acl);
 
@@ -654,7 +654,7 @@ static int ntfs_xattr_set_acl(struct user_namespace *mnt_userns,
 		return -EOPNOTSUPP;
 	}
 
-	if (!inode_owner_or_capable(mnt_userns, inode))
+	if (!inode_owner_or_capable(inode))
 		return -EPERM;
 
 	if (!value) {
@@ -671,7 +671,7 @@ static int ntfs_xattr_set_acl(struct user_namespace *mnt_userns,
 		}
 	}
 
-	err = ntfs_set_acl(mnt_userns, inode, acl, type);
+	err = ntfs_set_acl(inode, acl, type);
 
 release_and_out:
 	posix_acl_release(acl);
@@ -980,7 +980,7 @@ static noinline int ntfs_setxattr(const struct xattr_handler *handler,
 	     !memcmp(name, XATTR_NAME_POSIX_ACL_DEFAULT,
 		     sizeof(XATTR_NAME_POSIX_ACL_DEFAULT)))) {
 		err = ntfs_xattr_set_acl(
-			mnt_userns, inode,
+			&init_user_ns, inode,
 			name_len == sizeof(XATTR_NAME_POSIX_ACL_ACCESS) - 1
 				? ACL_TYPE_ACCESS
 				: ACL_TYPE_DEFAULT,
-- 
2.34.1

